README.md

```diff
@@ -5,13 +5,80 @@
+`Result<Value, Error>` values are either successful (wrapping `Value`) or failed (wrapping `Error`). This is similar to Swift’s native `Optional` type: `Success` is like `Some`, and `Failure` is like `None` except with an associated `ErrorType` value. The addition of an associated `ErrorType` allows errors to be passed along for logging or displaying to the user.
 
+Using this µframework instead of rolling your own `Result` type allows you to easily interface with other frameworks that also use `Result`.
 
 ## Use
 
-[API documentation](http://cocoadocs.org/docsets/Result/) is in the source.
+Use `Result` whenever an operation has the possibility of failure. Consider the following example of a function that tries to extract a `String` for a given key from a JSON `Dictionary`.
 
+```swift
+typealias JSONObject = [String:AnyObject]
+
+enum JSONError : ErrorType {
+    case NoSuchKey(String)
+    case TypeMismatch
+}
+
+func stringForKey(json: JSONObject, key: String) -> Result<String, JSONError> {
+    guard let value = json[key] else {
+        return .Failure(.NoSuchKey(key))
+    }
+    
+    if let value = value as? String {
+        return .Success(value)
+    }
+    else {
+        return .Failure(.TypeMismatch)
+    }
+}
+```
+
+This function provides a more robust wrapper around the default subscripting provided by `Dictionary`. Rather than return `AnyObject?`, it returns a `Result` that either contains the `String` value for the given key, or an `ErrorType` detailing what went wrong.
+
+One simple way to handle a `Result` is to deconstruct it using a `switch` statement.
+
+```swift
+switch stringForKey(json, key: "email") {
+
+case let .Success(email):
+    print("The email is \(email)")
+    
+case let .Failure(JSONError.NoSuchKey(key)):
+    print("\(key) is not a valid key")
+    
+case .Failure(JSONError.TypeMismatch):
+    print("Didn't have the right type")
+}
+```
+
+Using a `switch` statement allows powerful pattern matching, and ensures all possible results are covered. Swift 2.0 offers new ways to deconstruct enums like the `if-case` statement, but be wary as such methods do not ensure errors are handled.
+
+Other methods available for processing `Result` are detailed in the [API documentation](http://cocoadocs.org/docsets/Result/).
+
+## Result vs. Throws
+
+Swift 2.0 introduces error handling via throwing and catching `ErrorType`. `Result` accomplishes the same goal by encapsulating the result instead of hijacking control flow. The `Result` abstraction allows enables powerful functionality such as `map` and `flatMap`, making `Result` more composable than `throw`.
+
+Since dealing with APIs that throw is common, you can convert functions such functions into a `Result` by using the `materialize` method. Conversely, a `Result` can be used to throw an error by calling `dematerialize`. [Note: due to compiler issues, `materialize` is not currently available]
```

Comment:
It'd be great to see an example here and a note about what compiler issues are preventing this from moving forward.